src/CommunityToolkit.Maui.Analyzers/UseCommunityToolkitInitializationAnalyzer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CommunityToolkit.Maui.Analyzers;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class UseCommunityToolkitInitializationAnalyzer : DiagnosticAnalyzer
+{
+	public const string DiagnosticId = "MCT001";
+
+	const string category = "Initialization";
+	static readonly LocalizableString title = new LocalizableResourceString(nameof(Resources.InitializationErrorTitle), Resources.ResourceManager, typeof(Resources));
+	static readonly LocalizableString messageFormat = new LocalizableResourceString(nameof(Resources.InitalizationMessageFormat), Resources.ResourceManager, typeof(Resources));
+	static readonly LocalizableString description = new LocalizableResourceString(nameof(Resources.InitializationErrorMessage), Resources.ResourceManager, typeof(Resources));
+
+	static readonly DiagnosticDescriptor rule = new(DiagnosticId, title, messageFormat, category, DiagnosticSeverity.Error, isEnabledByDefault: true, description: description);
+
+	public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+
+	public override void Initialize(AnalysisContext context)
+	{
+		context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+		context.EnableConcurrentExecution();
+		context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ExpressionStatement);
+	}
+
+	static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+	{
+		var expressionStatement = (ExpressionStatementSyntax)context.Node;
+
+		if (expressionStatement.ToString().Contains("UseMauiApp<")
+			&& !expressionStatement.ToString().Contains(".UseMauiCommunityToolkit("))
+		{
+			var expression = GetInvocationExpressionSyntax(expressionStatement);
+			var diagnostic = Diagnostic.Create(rule, expression.GetLocation());
+			context.ReportDiagnostic(diagnostic);
+		}
+	}
+
+	static InvocationExpressionSyntax GetInvocationExpressionSyntax(SyntaxNode parent)
```

Comment:
This is needed because if the user has something like

```chsarp
builder = builder
	.UseMauiApp<App>();
	builder.ConfigureFonts(fonts =>
	{
		fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
		fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
	});
```

we can remove the `builder =` and deliver the right pieces to the `CodeFixProvider`